src/components/PeopleFilter/PeopleFilter.tsx

```diff
@@ -0,0 +1,138 @@
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import React from 'react';
+import { SearchLink } from '../SearchLink';
+
+export const PeopleFilters = React.memo(() => {
+  const centuriesValues = ['16', '17', '18', '19', '20'];
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex');
+
+  const updateSearchParams = (
+    params: { [key: string]: string | string[] | null },
+  ) => {
+    Object.entries(params).forEach(([key, value]) => {
+      if (!value) {
+        searchParams.delete(key);
+      } else if (Array.isArray(value)) {
+        searchParams.delete(key);
+        value.forEach(part => {
+          searchParams.append(key, part);
+        });
+      } else {
+        searchParams.set(key, value);
+      }
+    });
+
+    setSearchParams(searchParams);
+  };
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    updateSearchParams({ query: event.target.value || null });
+  };
+
+  const handleResetAllFilters = () => {
+    updateSearchParams({
+      query: null,
+      centuries: null,
+      sex: null,
+    });
+  };
+
+  return (
+    <nav className="panel">
+      <p className="panel-heading">Filters</p>
+
+      <p className="panel-tabs" data-cy="SexFilter">
+        <SearchLink
+          className={cn({ 'is-active': !sex })}
+          params={{ sex: null }}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          className={cn({ 'is-active': sex === 'm' })}
+          params={{ sex: 'm' }}
+        >
+          Male
+        </SearchLink>
+
+        <SearchLink
+          className={cn({ 'is-active': sex === 'f' })}
+          params={{ sex: 'f' }}
+        >
+          Female
+        </SearchLink>
```

Comment:
I think better to make one more component as PageNavLink and don't repeat the same code multiple times.

Reply:
u a right it's good idea